### PR TITLE
new Parametrizer tests for LieGroup, VectorSpace and Metric

### PR DESCRIPTION
**Pull request by SaitejaUtpala**
Following parametrized tests are added. 

**LieGroup**
- test_compose_point_inv_point_is_identity
- test_compose_inv_point_point_is_identity
- test_compose_point_identity_is_identity
- test_compose_identity_point_is_identity

**Vector Space**
- test_random_point_is_tangent
- test_to_tangent_is_projection

**Metric**
- test_dist_is_symmetric
- test_dist_is_positive
- test_squared_dist_is_positive
- test_inner_product_is_symmetric
- test_dist_is_norm_of_log
- test_dist_point_to_itself_is_zero

### CODE REVIEW

**Comment by SaitejaUtpala** [comment]
@ninamiolane  @nguigs. Let me if there are any changes that need to be done. Once it is done. I will write data generation classes and adapt existing manifolds

**Review by ninamiolane — COMMENTED** [review]
Design LGTM, but tests need to pass.

**Comment by SaitejaUtpala** [comment]
@ninamiolane. I will work on this today. Thanks!

**Comment by SaitejaUtpala** [comment]
@ninamiolane  can you review tests and their names. I will write data generation classes. Thanks!

File: tests/geometry_test_cases.py [hunk 1]
```diff
@@ -624,6 +735,48 @@ def test_exp_geodesic_ivp(
...
+            Point on the manifold.
+        point_b : array-like
+            Point on the manifold.
+        rtol : float
+            Relative tolerance to test this property.
+        atol : float
+            Absolute tolerance to test this property.
+        """
+        metric = self.metric(*metric_args)
+        sd_a_b = metric.dist(gs.array(point_a), gs.array(point_b))
```
Comment (ninamiolane) [inline tests/geometry_test_cases.py:755]: Very NIT: why `sd`? dist_a_b could be clearer.

**Review by ninamiolane — COMMENTED** [review]
LGTM! Some minor remarks + one comments about the use of the negative part to test dists are positive. I assume this is done for tolerance reason, but shouldn't there be a faster-easier way to test >= 0 ? Thank you.

File: tests/data_generation.py [hunk 2]
```diff
@@ -291,6 +291,85 @@ def _intrinsic_then_extrinsic_test_data(
...
+    ):
+        """Generate data to check that group exponential and logarithm are inverse.
+
+        Parameters
+        ----------
+        group_cls : LieGroup
+            Class of the group, i.e. a child class of LieGroup.
+        group_args_list : list
+            Arguments to pass to constructor of the Lie group.
+        n_tangent_vecs_list : list
```
Comment (ninamiolane) [inline tests/data_generation.py:305]: change docstring

File: tests/data_generation.py [hunk 3]
```diff
@@ -291,6 +291,85 @@ def _intrinsic_then_extrinsic_test_data(
...
```
Comment (ninamiolane) [inline tests/data_generation.py:330]: ditto

File: tests/data_generation.py [hunk 4]
```diff
@@ -291,6 +291,85 @@ def _intrinsic_then_extrinsic_test_data(
...
```
Comment (ninamiolane) [inline tests/data_generation.py:348]: ditto

File: tests/data_generation.py (hunk 2 above)
Comment (ninamiolane) [reply tests/data_generation.py:305]: also missing rtol and atol

File: tests/tests_geomstats/test_invariant_metric.py [hunk 5]
```diff
@@ -30,6 +29,7 @@ class TestInvariantMetric(RiemannianMetricTestCase, metaclass=Parametrizer):
     skip_test_log_then_exp = np_backend()
     skip_test_geodesic_bvp_belongs = True
     skip_test_exp_then_log = True
+    skip_test_dist_point_to_itself_is_zero = True
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_invariant_metric.py:32]: This does not work? :/ 
what is the error and did you raise an issue on it?

File: tests/tests_geomstats/test_minkowski.py [hunk 6]
```diff
@@ -259,6 +277,52 @@ def parallel_transport_bvp_is_isometry_test_data(self):
...
+                self.n_points_b_list,
+            )
+
+        def squared_dist_is_positive_test_data(self):
+            return self._squared_dist_is_positive_test_data(
+                self.metric_args_list,
+                self.space_list,
+                self.n_points_a_list,
+                self.n_points_b_list,
+                is_positive_atol=gs.atol * 10000,
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_minkowski.py:302]: Minkowski is a pseudo-Riemannian metric (it is actually the only one in the library), thus the squared dist can be negative, I would suggest removing this test (ignoring, as opposed to skipping).

**Review by ninamiolane — APPROVED** [review]
LGTM, except:
- docstrings
- minkowski negative distance can be ignored
See my comments

File: tests/tests_geomstats/test_invariant_metric.py (hunk 5 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_invariant_metric.py:32]: yeah.. sure I will raise it

File: tests/data_generation.py (hunk 2 above)
Comment (ninamiolane) [reply tests/data_generation.py:305]: change docstring: n_points instead of n_tangent_vecs

File: tests/data_generation.py (hunk 3 above)
Comment (ninamiolane) [reply tests/data_generation.py:334]: ditto

File: tests/data_generation.py (hunk 4 above)
Comment (ninamiolane) [reply tests/data_generation.py:356]: ditto

File: tests/data_generation.py [hunk 7]
```diff
@@ -946,6 +1146,142 @@ def _squared_dist_is_symmetric_test_data(
...
+    ):
+        """Generate data to check that the squared geodesic distance is symmetric.
+
+        Parameters
+        ----------
+        metric_args_list : list
+            List of arguments to pass to constructor of the metric.
+        space_list : list
+            List of spaces on which the metric is defined.
+        n_points_a_list : list
```
Comment (ninamiolane) [inline tests/data_generation.py:1263]: bad name

File: tests/geometry_test_cases.py [hunk 8]
```diff
@@ -645,6 +798,54 @@ def test_squared_dist_is_symmetric(self, metric_args, point_a, point_b, rtol, at
...
+
+        Parameters
+        ----------
+        metric_args : tuple
+            Arguments to pass to constructor of the metric.
+        tangent_vec_a : array-like
+            Tangent vector to the manifold at base_point.
+        tangent_vec_b : array-like
+            Tangent vector to the manifold at base_point.
+        point_a : array-like
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply tests/geometry_test_cases.py:851]: why did this get resolved? it is not resolved.

**Review by ninamiolane — APPROVED** [review]
Please check the docstrings again (parameter names, specifically).

File: tests/geometry_test_cases.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (ninamiolane) [reply tests/geometry_test_cases.py:851]: adapt the docstring: 
- point_a and point_b are not inputs of this function, thus they should not appear in the parameters documentations.
- base_point is a parameter of this function, thus it should appear in the parameters documentation

check the other docstrings and let me know when I can review?

**Review by ninamiolane — CHANGES_REQUESTED** [review]
Remaining docstrings issues

**Comment by SaitejaUtpala** [comment]
> Remaining docstrings issues
really sorry! that I keep missing these things!

**Comment by ninamiolane** [comment]
No worries, it is really hard to check all of them (and there are a lot). Thank you, and thanks for this giant PR again. I'll let you merge if the failing test is not an issue.

**Comment by SaitejaUtpala** [comment]
> No worries, it is really hard to check all of them (and there are a lot). Thank you, and thanks for this giant PR again. I'll let you merge if the failing test is not an issue.

I am skipping these tests in 32 bit backends. Thanks!

**Pull request merged**